controllers/v1/reportsController.js

```diff
@@ -868,61 +911,25 @@ module.exports = class Reports {
 
-        Promise.all([submissionDocument, evaluationFrameworksDocuments]).then(submissionAndEvaluationFrameworksDocuments => {
-          let submissionDocument = submissionAndEvaluationFrameworksDocuments[0];
-          let evaluationFrameworksDocuments = submissionAndEvaluationFrameworksDocuments[1];
-
-          let evaluationNameObject = {};
-          evaluationFrameworksDocuments.forEach(singleDocument => {
-            singleDocument.themes.forEach(singleTheme => {
-              singleTheme.children && singleTheme.children.forEach(subThemes => {
-                subThemes.children && subThemes.children.forEach(singleSubTheme => {
-                  singleSubTheme.criteria.forEach(singleCriteria => {
-                    evaluationNameObject[singleCriteria.toString()] = {
-                      themeName: singleTheme.name,
-                      aoiName: subThemes.name,
-                      indicatorName: singleSubTheme.name
-                    };
-                  });
-                })
-              })
-            });
-          });
-
-          if (!submissionDocument && !submissionDocument[0].criterias.length) {
-            return resolve({
-              status: 404,
-              message: "No submissions found for given params."
-            });
-          }
-          else {
-            submissionDocument[0].criterias.forEach(submissionCriterias => {
-              let levels = Object.values(submissionCriterias.rubric.levels);
-
-              if (submissionCriterias._id) {
-                let criteriaReportObject = {
-                  "Theme Name": evaluationNameObject[submissionCriterias._id]
-                    ? evaluationNameObject[submissionCriterias._id].themeName
-                    : "",
-                  "AoI Name": evaluationNameObject[submissionCriterias._id]
-                    ? evaluationNameObject[submissionCriterias._id].aoiName
-                    : "",
-                  "Level 1": levels.find(level => level.level == "L1").description,
-                  "Level 2": levels.find(level => level.level == "L2").description,
-                  "Level 3": levels.find(level => level.level == "L3").description,
-                  "Level 4": levels.find(level => level.level == "L4").description,
-                  "Score": submissionCriterias.score
-                    ? submissionCriterias.score
-                    : "NA"
-                };
-                input.push(criteriaReportObject);
-              }
-            });
+        submissionDocument.criterias && submissionDocument.criterias.forEach(submissionCriterias => {
+          let levels = Object.values(submissionCriterias.rubric.levels);
+
+          if (submissionCriterias._id) {
+            let criteriaReportObject = {
+              "Path To Criteria": arr[submissionCriterias._id.toString()] ? arr[submissionCriterias._id.toString()].parentPath : "",
+              "Level 1": levels.find(level => level.level == "L1").description,
```

Comment:
Please can we make this dynamic @aman-tunerlabs

Reply:
Done .